nncf/quantization/init_precision.py

```diff
@@ -168,33 +169,33 @@ def apply_init(self):
-
-        chosen_config_per_layer = self.choose_configuration(configuration_metric, bits_configurations,
-                                                            traces_per_layer.get_order_of_traces())
-        self.set_chosen_config(chosen_config_per_layer)
-        ordered_metric_per_layer = self.get_metric_per_layer(chosen_config_per_layer, perturbations,
-                                                             traces_per_layer)
-        if is_debug():
-            hawq_debugger = HAWQDebugger(bits_configurations, perturbations,
-                                         weight_observers, traces_per_layer, self._bits)
-            hawq_debugger.dump_metric(configuration_metric)
-            hawq_debugger.dump_avg_traces()
-            hawq_debugger.dump_density_of_quantization_noise()
-            hawq_debugger.dump_perturbations_ratio()
-            hawq_debugger.dump_bitwidth_graph(self._algo, self._model)
-
-        self._model.rebuild_graph()
-        str_bw = [str(element) for element in self.get_bitwidth_per_scope()]
-        nncf_logger.info('\n'.join(['\n\"bitwidth_per_scope\": [', ',\n'.join(str_bw), ']']))
-
-        self._model.to(original_device)
-        return ordered_metric_per_layer
+            warnings.warn('All bits configurations are incompatible with HW Config!', RuntimeWarning)
+        else:
+            perturbations, weight_observers = self.calc_quantization_noise()
+
+            configuration_metric = self.calc_hawq_metric_per_configuration(bits_configurations, perturbations,
+                                                                            traces_per_layer, self._init_device)
+
+            chosen_config_per_layer = self.choose_configuration(configuration_metric, bits_configurations,
+                                                                traces_per_layer.get_order_of_traces())
+            self.set_chosen_config(chosen_config_per_layer)
+            ordered_metric_per_layer = self.get_metric_per_layer(chosen_config_per_layer, perturbations,
+                                                                    traces_per_layer)
+            if is_debug():
+                hawq_debugger = HAWQDebugger(bits_configurations, perturbations,
+                                                weight_observers, traces_per_layer, self._bits)
+                hawq_debugger.dump_metric(configuration_metric)
+                hawq_debugger.dump_avg_traces()
+                hawq_debugger.dump_density_of_quantization_noise()
+                hawq_debugger.dump_perturbations_ratio()
+                hawq_debugger.dump_bitwidth_graph(self._algo, self._model)
+
+                self._model.rebuild_graph()
+                str_bw = [str(element) for element in self.get_bitwidth_per_scope()]
+                nncf_logger.info('\n'.join(['\n\"bitwidth_per_scope\": [', ',\n'.join(str_bw), ']']))
+
+                self._model.to(original_device)
+                return ordered_metric_per_layer
```

Comment:
What is this whitespace change all about?

Reply:
@vshampor, As far as I can see, Anastasia moved the whole block to the right under else condition. 
If all bit-width settings, generated by HAWQ, are incompatible with HW config, precision initialization will do nothing.

Definitely, this behavior should be fixed. By default, the error should be raised. But looks like it tests for manual HAWQ will fail. @asenina, am I right? If so, I can fix them later and this change is OK.

Reply:
why warnings? just to not fail tests? I'd prefer disabling/skipping tests then. The right behavior is an error, as we discussed with the team. If a user wants warnings then he should specify `treat_warning_as_errors`=False, which is True, by default.